python/cudf/bindings/groupby.pyx

```diff
@@ -0,0 +1,462 @@
+                    err = gdf_group_by_avg(
+                        ncols,
+                        vector_cols.data(),
+                        col_agg,
+                        out_col_indices,
+                        vector_out_col_values.data(),
+                        out_col_agg,
+                        ctx
+                    )
+                elif agg_type == 'min':
+                    err = gdf_group_by_min(
+                        ncols,
+                        vector_cols.data(),
+                        col_agg,
+                        out_col_indices,
+                        vector_out_col_values.data(),
+                        out_col_agg,
+                        ctx
+                    )
+                elif agg_type == 'max':
+                    err = gdf_group_by_max(
+                        ncols,
+                        vector_cols.data(),
+                        col_agg,
+                        out_col_indices,
+                        vector_out_col_values.data(),
+                        out_col_agg,
+                        ctx
+                    )
+                elif agg_type == 'count':
+                    err = gdf_group_by_count(
+                        ncols,
+                        vector_cols.data(),
+                        col_agg,
+                        out_col_indices,
+                        vector_out_col_values.data(),
+                        out_col_agg,
+                        ctx
+                    )
+                elif agg_type == 'sum':
+                    err = gdf_group_by_sum(
+                        ncols,
+                        vector_cols.data(),
+                        col_agg,
+                        out_col_indices,
+                        vector_out_col_values.data(),
+                        out_col_agg,
+                        ctx
+                    )
+
+        check_gdf_error(err)
+
+        num_row_results = out_col_agg.size
+
+        # NVStrings columns are not the same going in as coming out but we
+        # can't create entire memory views otherwise multiple objects will
+        # try to free the memory
+        for i, col in enumerate(out_col_values_series):
+            if col.dtype == np.dtype("object") and len(col) > 0:
+                import nvcategory
+                nvcat_ptr = int(<uintptr_t>vector_out_col_values[i].dtype_info.category)
+                nvcat_obj = None
+                if nvcat_ptr:
+                    nvcat_obj = nvcategory.bind_cpointer(nvcat_ptr)
+                    nvstr_obj = nvcat_obj.to_strings()
+                else:
+                    import nvstrings
+                    nvstr_obj = nvstrings.to_device([])
+                out_col_values_series[i]._column._data = nvstr_obj
+                out_col_values_series[i]._column._nvcategory = nvcat_obj
+        if out_col_agg_series.dtype == np.dtype("object") and \
+                len(out_col_agg_series) > 0:
+            import nvcategory
```

Comment:
Isn't this the exact same code block as above (lines 227 to 238)? If so maybe try making them a function?

Reply:
@dantegd this is fixed in #1492